testBuilderUsesDefaultSSLContext fails due to a wrong exception message

This test failed today of the above test while running on java11, which may be relevant:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java11,nodes=virtual&&linux/168/console

```
16:26:03 FAILURE 0.58s J2 | RestClientBuilderIntegTests.testBuilderUsesDefaultSSLContext <<< FAILURES!
16:26:03    > Throwable #1: java.lang.AssertionError: 
16:26:03    > Expected: a string containing "General SSLEngine problem"
16:26:03    >      but: was "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target"
16:26:03    > 	at __randomizedtesting.SeedInfo.seed([40A8A6346B9B56FA:78D0B8F8B871D88F]:0)
16:26:03    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
16:26:03    > 	at org.elasticsearch.client.RestClientBuilderIntegTests.testBuilderUsesDefaultSSLContext(RestClientBuilderIntegTests.java:82)
16:26:03    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
16:26:03    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
16:26:03    > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
16:26:03    > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
16:26:03    > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

The SmokeTestMonitoringWithSecurityIT#testHTTPExporterWithSSL() test looks also to be failing because of the jdk11 bug:

```
17:43:20 [2018-07-11T17:43:09,610][ERROR][o.e.x.m.e.h.VersionHttpResource] failed to verify minimum version [7.0.0-alpha1] on the [xpack.monitoring.exporters._http] monitoring cluster
17:43:20 java.lang.RuntimeException: null
17:43:20 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:951) ~[elasticsearch-rest-client-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:227) ~[elasticsearch-rest-client-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:294) ~[elasticsearch-rest-client-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.VersionHttpResource.doCheckAndPublish(VersionHttpResource.java:62) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpResource.checkAndPublish(HttpResource.java:156) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.MultiHttpResource.doCheckAndPublish(MultiHttpResource.java:64) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpResource.checkAndPublish(HttpResource.java:156) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpResource.blockUntilCheckAndPublish(HttpResource.java:132) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpResource.checkAndPublishIfDirty(HttpResource.java:116) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpExporter.isExporterReady(HttpExporter.java:639) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpExporter.openBulk(HttpExporter.java:645) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.http.HttpExporter.openBulk(HttpExporter.java:67) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.openBulk(Exporters.java:115) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.export(Exporters.java:186) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.xpack.monitoring.MonitoringService$MonitoringExecution$1.doRun(MonitoringService.java:227) [x-pack-monitoring-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
17:43:20 	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
17:43:20 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:624) [elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
17:43:20 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
17:43:20 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
17:43:20 	at java.lang.Thread.run(Thread.java:834) [?:?]
17:43:20 Caused by: java.lang.NullPointerException
17:43:20 	at sun.security.ssl.SSLSessionImpl.getLocalPrincipal(SSLSessionImpl.java:661) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.client.DefaultAsyncUserTokenHandler.getUserToken(DefaultAsyncUserTokenHandler.java:85) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.client.MainClientExec.responseCompleted(MainClientExec.java:377) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:168) ~[?:?]
17:43:20 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436) ~[?:?]
17:43:20 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:121) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[?:?]
17:43:20 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588) ~[?:?]
17:43:20 	... 1 more
```

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/170/console

--- a/client/rest/src/test/java/org/elasticsearch/client/RestClientBuilderIntegTests.java
+++ b/client/rest/src/test/java/org/elasticsearch/client/RestClientBuilderIntegTests.java
@@ -30,6 +30,7 @@ import org.junit.BeforeClass;
 
 import javax.net.ssl.KeyManagerFactory;
 import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
 import javax.net.ssl.TrustManagerFactory;
 import java.io.IOException;
 import java.io.InputStream;
@@ -38,6 +39,7 @@ import java.net.InetSocketAddress;
 import java.security.KeyStore;
 
 import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.fail;
@@ -72,9 +74,6 @@ public class RestClientBuilderIntegTests extends RestClientTestCase {
     }
 
     public void testBuilderUsesDefaultSSLContext() throws Exception {
-        assumeFalse("Due to bug inside jdk, this test can't momentarily run with java 11. " +
-                "See: https://github.com/elastic/elasticsearch/issues/31940",
-            System.getProperty("java.version").contains("11"));
         final SSLContext defaultSSLContext = SSLContext.getDefault();
         try {
             try (RestClient client = buildRestClient()) {
@@ -82,7 +81,7 @@ public class RestClientBuilderIntegTests extends RestClientTestCase {
                     client.performRequest(new Request("GET", "/"));
                     fail("connection should have been rejected due to SSL handshake");
                 } catch (Exception e) {
-                    assertThat(e.getMessage(), containsString("General SSLEngine problem"));
+                    assertThat(e, instanceOf(SSLHandshakeException.class));
                 }
             }
 

--- a/client/rest/src/test/java/org/elasticsearch/client/RestClientBuilderIntegTests.java
+++ b/client/rest/src/test/java/org/elasticsearch/client/RestClientBuilderIntegTests.java
@@ -38,7 +38,6 @@ import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.security.KeyStore;
 
-import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;

--- a/x-pack/qa/smoke-test-plugins-ssl/src/test/java/org/elasticsearch/smoketest/SmokeTestMonitoringWithSecurityIT.java
+++ b/x-pack/qa/smoke-test-plugins-ssl/src/test/java/org/elasticsearch/smoketest/SmokeTestMonitoringWithSecurityIT.java
@@ -127,7 +127,6 @@ public class SmokeTestMonitoringWithSecurityIT extends ESIntegTestCase {
         return monitoringUsage.get().getExporters().isEmpty() == false;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/31940")
     public void testHTTPExporterWithSSL() throws Exception {
         // Ensures that the exporter is actually on
         assertBusy(() -> assertThat("[_http] exporter is not defined", getMonitoringUsageExportersDefined(), is(true)));

